syncd/ConcurrentQueue.h

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mutex>
+#include <queue>
+
+#include "swss/logger.h"
+
+namespace syncd
+{
+    template <class T>
+    class ConcurrentQueue
+    {
+        public:
+
+            explicit ConcurrentQueue(
+                    size_t queueSizeLimit = m_unlimited);
+
+            virtual ~ConcurrentQueue();
+
+            bool enqueue(
+                    const T& val);
+
+            bool dequeue(
+                    T* valOut);
+
+            size_t getQueueSize();
+
+            bool empty();
+
+        private:
+
+            // If queue size is specified 0, that will mean there is no limit on queue
+            // size.
+            static constexpr size_t m_unlimited = 0;
```

Comment:
if it's a static constexpr definition maybe you could make it capital letters as size_t UNLIMITED = 0;